src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -153,7 +155,17 @@ protected function componentString(string $component, array $attributes)
 
         [$data, $attributes] = $this->partitionDataAndAttributes($class, $attributes);
 
-        return " @component('{$class}', [".$this->attributesToString($data->all()).'])
+        // If the component doesn't exists as a class we'll assume it's a class-less component
+        // and pass the component as a view parameter to the data collection bag.
+        if (! class_exists($class)) {
```

Comment:
The only thing that worries me is how to distinguish between a user making a typo in a class name, and actually wanting to not have a class.

Reply:
That's a really big edge case imo. Plus I guess you'll immediately see it?